php/admin/class-onboarding.php

```diff
@@ -0,0 +1,392 @@
+						'<a href="' . get_delete_post_link( $example_post_id ) . '" class="trash">',
+						'</a>'
+					)
+				);
+				?>
+			</p>
+			<span class="pointer">☝️</span>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add Fields message.
+	 */
+	public function show_add_fields_notice() {
+		$post  = get_post();
+		$block = new Block( $post->ID );
+
+		/**
+		 * We add 4 fields to our Example Block in add_dummy_data().
+		 */
+		if ( count( $block->fields ) > 4 ) {
+			return;
+		}
+		?>
+		<div class="block-lab-add-fields block-lab-notice">
+			<h2>🧐 <?php esc_html_e( 'Try adding a field.', 'block-lab' ); ?></h2>
+			<p><?php esc_html_e( 'Fields let you define the options you see when adding your block to a post.', 'block-lab' ); ?></p>
+			<span class="pointer">👈</span>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add Fields message.
+	 */
+	public function show_publish_notice() {
+		$post  = get_post();
+		$block = new Block( $post->ID );
+
+		/**
+		 * We add 4 fields to our Example Block in add_dummy_data().
+		 */
+		if ( count( $block->fields ) > 4 ) {
+			return;
+		}
+		?>
+		<div class="block-lab-publish block-lab-notice">
+			<h2>🧪 <?php esc_html_e( 'Time to experiment!', 'block-lab' ); ?></h2>
+			<p class="intro"><?php esc_html_e( 'Choose an icon, change the category, and investigate a few different field types.', 'block-lab' ); ?></p>
+			<p>
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// translators: Placeholders are <strong> html tags.
+						__( 'When you\'re ready, save your block by pressing %1$sPublish%2$s.', 'block-lab' ),
+						'<strong>',
+						'</strong>'
+					)
+				);
+				?>
+			</p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add to Post message.
+	 */
+	public function show_add_to_post_notice() {
+		$post = get_post();
+
+		if ( ! $post || ! isset( $post->post_name ) || empty( $post->post_name ) ) {
+			return;
+		}
+
+		if ( 'publish' !== $post->post_status ) {
+			return;
+		}
+
+		$template = block_lab_locate_template( 'blocks/block-' . $post->post_name . '.php', '', true );
+
+		if ( ! $template ) {
+			return;
+		}
+		?>
+		<div class="block-lab-add-to-block block-lab-notice notice notice-large is-dismissible">
+			<h2>🚀 <?php esc_html_e( 'Only one thing left to do!', 'block-lab' ); ?></h2>
+			<p class="intro"><?php esc_html_e( 'You\'ve created a new block, and added a block template. Well done!', 'block-lab' ); ?></p>
+			<p><?php esc_html_e( 'All that\'s left is to add your block to a post.', 'block-lab' ); ?></p>
+			<a href="<?php echo esc_attr( admin_url( 'post-new.php' ) ); ?>" class="button">
+				<?php esc_html_e( 'Add New Post', 'block-lab' ); ?>
+			</a>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Create a dummy starter block when the plugin is activated for the first time.
+	 */
+	public function add_dummy_data() {
+		// Note: wp_count_posts() does not work here.
+		$blocks = get_posts(
+			array(
+				'post_type'   => block_lab()->block_post->slug,
+				'numberposts' => '-1',
```

Comment:
Would it be possible to set some kind of limit?

Reply:
1 aught to work!

Reply:
Fixed in 86fbdb3.

Reply:
Nice!